checkov/terraform/checks/graph_checks/azure/AzureAKSclusterAzureCNIEnabled.yaml

```
@@ -0,0 +1,20 @@
+metadata:
+  id: "CKV2_AZURE_29"
+  name: "Ensure AKS cluster has Azure CNI networking enabled"
+  category: "GENERAL_SECURITY"
+
+definition:
+  and:
+
+    - cond_type: "attribute"
+      resource_types: "azurerm_kubernetes_cluster"
+      attribute: "network_profile.network_plugin"
+      operator: "exists"
+
+    - cond_type: "attribute"
+      resource_types: "azurerm_kubernetes_cluster"
+      attribute: "network_profile.network_plugin"
+      operator: "within"
+      value: 
+        - "azure"
+        - "Azure"
```

```suggestion
    cond_type: "attribute"
    resource_types: "azurerm_kubernetes_cluster"
    attribute: "network_profile.network_plugin"
    operator: "within"
    value: 
      - "azure"
      - "Azure"
```
I can think you don't need the exist block, this should achieve the same.

---

Yep, tested. You're right 🙌🏻

I've updated it.